cmd/plugin/plugin.go

```diff
@@ -18,39 +18,36 @@ var (
 	log     = logf.Log
 )
 
-func main() {
-	root := &cobra.Command{
-		Use:   "dns",
-		Short: "DNS Operator command line utility",
-		Long:  "DNS Operator command line utility",
-		PersistentPreRun: func(cmd *cobra.Command, args []string) {
-			logf.SetLogger(zap.New(zap.UseDevMode(verbose), zap.WriteTo(os.Stdout)))
-			cmd.SetContext(context.Background())
-		},
-	}
-
-	root.SetArgs(os.Args[1:])
+var rootCMD = &cobra.Command{
+	Use:   "dns",
+	Short: "DNS Operator command line utility",
+	Long:  "DNS Operator command line utility",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		logf.SetLogger(zap.New(zap.UseDevMode(verbose), zap.WriteTo(os.Stdout)))
+		cmd.SetContext(context.Background())
+	},
+}
 
-	root.PersistentFlags().BoolVarP(&verbose, "verbose", "v", true, "verbose output")
+var versionCMD = &cobra.Command{
+	Use:   "version",
+	Short: "Print the version number of cli",
+	Long:  "Print the version number of cli",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		fmt.Printf("dns cli version: %s (%s)\n", version, gitSHA)
+		return nil
+	},
+}
 
-	root.AddCommand(versionCommand())
-	root.AddCommand(deleteOwnerCommand())
-	root.AddCommand(getZoneRecordsCommand())
-	root.AddCommand(secretGenerationCommand())
+func init() {
+	rootCMD.SetArgs(os.Args[1:])
+	rootCMD.PersistentFlags().BoolVarP(&verbose, "verbose", "v", true, "verbose output")
 
-	if err := root.Execute(); err != nil {
-		os.Exit(1)
-	}
+	rootCMD.AddCommand(versionCMD, deleteOldTXTCMD, getZoneRecordsCMD)
+	rootCMD.AddCommand(generateSecretCMD, deleteOwnerCMD)
```

Comment:
I feel a bit odd about this one. Why not add them all in same call? Do the grouping mean something?

Reply:
grouping means nothing. One command it is